Ensure that the app updates the comment on PR by a target user

dheerajodha opened the pull request.

asvoboda committed:
Add rate limit metrics (#16)

asvoboda committed:
Client caching as config option (#14)

svc-excavator-bot committed:
Excavator: Update godel to latest version (#11)

* Excavator: Update godel to latest version

* run godel format

bluekeyes committed:
Fix ordering of installation ID middleware (#17)

This must come first in the stack so that the value is available to
the metrics middleware added by users.

akerl committed:
Fix example functionality (#19)

This fixes the example API call, cleans up the logging, and adds a check
to ensure it doesn’t loop forever on its own comments

bluekeyes committed:
Add ability to load config from the environment (#20)

In certain platforms, setting application secrets is easier with
environment variables. Add a new function that can use environment
variables to override configuration values loaded from other places,
like a file.

bluekeyes committed:
Support custom responses from event handlers (#21)

This introduces two new mechanisms for customizing responses:

  * Clients can configure a response callback to send custom responses
    for all events.
  * Individual handlers may call SetResponder() to send custom responses
    in specific situations.

While there should be little need to customize responses, I believe some
combination of these two options will satisfy most use cases. Note that
introducing the new callback changes the signature of NewEventDispatcher
and some other types; most applications are not using these directly.

On the other hand, I chose to introduce SetResponder() to avoid breaking
the EventHandler interface, even though it is more idiomatic to either
provide the handler with access to the request and response writer or
allow it to return more information.

bluekeyes committed:
Add ValidationError type (#22)

When a payload fails validation, due to missing headers, an invalid
signature, or other reasons, create a ValidationError so that error
handlers can identify this case. Modify the default handler to return a
400 error in this case instead of a 500 or 202 response.

domsu committed:
Fixed nil pointer dereference (#23)

bluekeyes committed:
Fix middleware application race condition (#24)

We occasionally see a nil pointer exception that suggests the "next"
RoundTripper in the client middleware chain is nil. In reviewing the
middleware code, I noticed that the use of append() at client creation
time could modify the global slice associated with the creator instance
without a lock. This is definitely a race condition, and could be the
cause of the observed exception (but I have no direct evidence.)

Change the application function to take individual slices and iterate
over them directly, rather than creating a single slice.

svc-excavator-bot committed:
Excavator: Update godel to latest version (#27)

bluekeyes committed:
Add constraint for bradleyfalzon/ghinstallation (#29)

As of 1.0.0, this library includes a breaking API change and appears to
require that we adopt modules to be compatible with the go-github
imports.

bluekeyes committed:
Only set cache control on valid responses (#30)

If the delegate RoundTripper returns an error and no response, do not
try to access the response headers.

asvoboda committed:
Build with go 1.13.4 (#31)

svc-excavator-bot committed:
Excavator: Update godel to latest version (#32)

Co-authored-by: svc-excavator-bot <svc-excavator-bot@palantir.com>

bluekeyes committed:
Add InitializeResponder to improve testing (#34)

Allow code other than the default dispatcher to initialize a request
context for use with Set/GetResponder.

ajlake committed:
Convert to go mod (#35)

bluekeyes committed:
Upgrade google/go-github to v31 (#36)

bluekeyes committed:
Add support for asynchronous dispatch (#37)

Introduce a configurable scheduler option for NewEventDispatcher and add
two asynchronous schedulers, one that is unlimited and one that uses a
bounded queue and a fixed number of workers.

For many applications, switching to asynchronous handling will be as
easy as adding this new option when creating the event dispatcher. Other
applications will need to provide a custom ContextDeriver to copy values
from the request context into the detached handler context.

As a result of error handling code introduced as part of this change,
go-githubapp now requires at least Go 1.13.

IainSteers committed:
Fix typos in org client example in README (#38)

bluekeyes committed:
Fix typo in oauth2 error handler log (#40)

svc-excavator-bot committed:
Excavator: Update godel to latest version (#41)

Co-authored-by: svc-excavator-bot <svc-excavator-bot@palantir.com>

bluekeyes committed:
Update google/go-github to v32 (#43)

svc-excavator-bot committed:
Excavator: Update godel to latest version (#44)

Co-authored-by: svc-excavator-bot <svc-excavator-bot@palantir.com>

jmcampanini committed:
add ability to configure Timeout for http.Client (#48)

* centralize all http.Client creation

* set the timeout from the clientCreator

* add the configuration method

* add example and readme

* lint

* Update client_creator.go

bluekeyes committed:
Add cache status to GitHub request logs (#50)

bluekeyes committed:
Add event.dropped and event.age metrics (#52)

These provide more insight into the queued asynchronous scheduler.

bluekeyes committed:
Add error callbacks that emit metrics (#53)

These are non-default because they require a metrics registry, but the
default handlers are now implemented by passing a nil registry to the
metric version.

Note that this contains an API break in the AsyncErrorCallback type, but
I don't expect there to be many (any?) custom implementations.

fawind committed:
Apply caching middleware for token client (#57)

svc-excavator-bot committed:
Excavator: Update godel to latest version (#55)

Co-authored-by: svc-excavator-bot <svc-excavator-bot@palantir.com>

asvoboda committed:
Add policy.yml (#58)

jetersen committed:
Add GetByRepository for InstallationsService (#61)

bluekeyes committed:
Upgrade google/go-github to v33 (#63)

svc-excavator-bot committed:
Excavator: Updates godel plugins to latest versions (#60)

Co-authored-by: svc-excavator-bot <svc-excavator-bot@palantir.com>

svc-excavator-bot committed:
Excavator: Update godel to latest version (#64)

Co-authored-by: svc-excavator-bot <svc-excavator-bot@palantir.com>

svc-excavator-bot committed:
Excavator: Manage go module dependencies (#65)

Co-authored-by: svc-excavator-bot <svc-excavator-bot@palantir.com>

svc-excavator-bot committed:
Excavator: Manage go module dependencies (#68)

Co-authored-by: svc-excavator-bot <svc-excavator-bot@palantir.com>

asvoboda committed:
Enable codeql vuln scanning (#70)

andygrunwald committed:
Remove `public_url` from example configuration (#71)

The key `server.public_url` is not in use in any of this application.
It is removed for simplicity reasons

svc-excavator-bot committed:
Excavator: Update godel to latest version (#74)

Co-authored-by: svc-excavator-bot <svc-excavator-bot@palantir.com>

svc-excavator-bot committed:
Excavator: Removes use of Bintray resolvers from repository (#77)

Co-authored-by: svc-excavator-bot <svc-excavator-bot@palantir.com>

andygrunwald committed:
Upgrade google/go-github to v35 (#81)

svc-excavator-bot committed:
Excavator: Removes legacy format-plugin.yml configuration (#82)

svc-excavator-bot committed:
Excavator: Update godel to latest version (#79)

svc-excavator-bot committed:
Excavator: Update godel to latest version (#85)

bluekeyes committed:
Update google/go-github to v36 (#86)

bluekeyes committed:
Use GitHub Actions for CI (#87)

bluekeyes committed:
Update google/go-github to v37 (#88)

svc-excavator-bot committed:
Excavator: Manage go module dependencies (#90)

bluekeyes committed:
Add package for loading application config (#89)

The appconfig package defines a standard (but customizable) flow for
loading repository-level configuration:

1. Try a list of paths in the repository, in order
2. If a path exists, check if it contains a remote reference
    i. If it does, follow the reference
   ii. Otherwise, return the content
3. If no paths exist, try a different list of paths in a owner-default
   (e.g. .github) repository

The package doesn't require the configuration to have any format (it
returns raw bytes), but by default assumes that remote references are
encoded as YAML (this is customizable.)

This will hopefully make it easier to implement a consistent config
loading mechanism in our apps.

svc-excavator-bot committed:
Excavator: Manage go version (#91)

Commit:
Add support for refresh tokens (#92)

Add methods to be able to create token client from oauth2.TokenSource
This allows to use expiring user to server access tokens.

Signed-off-by: Torsten Walter <torsten.walter@syncier.com>

bluekeyes committed:
Fix formatting error in caching_client_creator.go (#93)

bluekeyes committed:
Fill in empty remote Git refs with default branch (#95)

This requires an extra API call, but ensures callers have accurate
information in the Source field. Also document the format of the Source
field and make the remote path optional as well.

bluekeyes committed:
Fix YAMLRemoteRefParser with empty input (#94)

Do not treat empty input as an invalid remote reference, since apps may
use an empty file as an on/off switch. Also add some tests for parsing,
since there are several edge cases now.

asvoboda committed:
Update go-github to v38 (#96)

asvoboda committed:
Support remote references in default config (#99)

asvoboda committed:
Update ghinstallation to v2 (#100)

bluekeyes committed:
Add WithTransport option to set TLS config (#102)

Users can provide an http.Transport instance with non-default TLS,
proxy, or timeout settings.

svc-excavator-bot committed:
Excavator: Manage go module dependencies (#98)

bluekeyes committed:
Fix nil pointer panic when using default transport (#105)

This was a regression introduced by the WithTransport option.

asvoboda committed:
Update go-github to v39 (#106)

svc-excavator-bot committed:
Excavator: Manage go module dependencies (#107)

asvoboda committed:
Update dependencies to latest versions (#110)

wwong committed:
Bump go-github to v40 (#115)

svc-excavator-bot committed:
Excavator: Manage go module dependencies (#114)

jamestoyer committed:
Add Dependabot dependency management (#117)

svc-excavator-bot committed:
Excavator: Manage go module dependencies (#116)

svc-excavator-bot committed:
Excavator: Update godel to latest version (#119)

svc-excavator-bot committed:
Excavator: Updates godel plugins to latest versions (#118)

svc-excavator-bot committed:
Excavator: Manage go module dependencies (#120)

dependabot[bot] committed:
Bump github.com/rs/zerolog from 1.26.0 to 1.26.1 (#123)

Bumps [github.com/rs/zerolog](https://github.com/rs/zerolog) from 1.26.0 to 1.26.1.
- [Release notes](https://github.com/rs/zerolog/releases)
- [Commits](https://github.com/rs/zerolog/compare/v1.26.0...v1.26.1)

---
updated-dependencies:
- dependency-name: github.com/rs/zerolog
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

svc-excavator-bot committed:
Excavator: Manage go version (#111)

andygrunwald committed:
Upgrade github.com/google/go-github from v40 to v41 (#125)

svc-excavator-bot committed:
Excavator: Manage go module dependencies (#124)

asvoboda committed:
Upgrade ghinstallation to latest (#126)

svc-excavator-bot committed:
Excavator: Update godel to latest version (#128)

svc-excavator-bot committed:
Excavator: Manage go module dependencies (#127)

svc-excavator-bot committed:
Excavator: Updates godel plugins to latest versions (#132)

svc-excavator-bot committed:
Excavator: Update godel to latest version (#130)

svc-excavator-bot committed:
Excavator: Upgrade to Go 1.17 (#131)

svc-excavator-bot committed:
Excavator: Updates godel plugins to latest versions (#134)

svc-excavator-bot committed:
Excavator: Manage go version (#133)

bluekeyes committed:
Use Go version from .palantir/go-version file (#136)

This opts us in to automatic version updates via Excavator and avoids
having to track the version in two places.

bluekeyes committed:
Update google/go-github to v42 (#137)

bluekeyes committed:
Add logging for client request and response bodies (#135)

Most useful for GraphQL queries, since there is no easy way to see the
actual query content otherwise. Added additional logging for
completeness. I don't think it will be useful most of the time, but
could be helpful to implement verbose modes or similar.

svc-excavator-bot committed:
Excavator: Update godel to latest version (#138)

bluekeyes committed:
Update google/go-github to v43 (#139)

svc-excavator-bot committed:
Excavator: Update godel to latest version (#140)

svc-excavator-bot committed:
Excavator: Manage go version (#142)

svc-excavator-bot committed:
Excavator: Updates godel plugins to latest versions (#141)

dependabot[bot] committed:
Bump actions/checkout from 2 to 3 (#143)

Bumps [actions/checkout](https://github.com/actions/checkout) from 2 to 3.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

svc-excavator-bot committed:
Excavator: Update godel to latest version (#144)

bluekeyes committed:
Improve handler panic reporting (#145)

Stack traces for handler panics were lost, making it hard to determine
where the panic actually happened. Now, stack traces are captured by a
special error type so that handlers can print them. By default, nothing
actually changes, but I plan to extend the default configuration of
go-baseapp so that these errors are formatted with stack traces in logs.

I copied the error code from my implementation in `bluekeyes/hatpear` so
that they'll share the same interface. The `pkg/errors` package has a
similar interface, but uses package-specific types. Using standard
library types seemed preferrable, even though we have a dependency on
`pkg/errors` already.

svc-excavator-bot committed:
Excavator: Update godel to latest version (#146)

svc-excavator-bot committed:
Excavator: Updates godel plugins to latest versions (#147)

svc-excavator-bot committed:
Excavator: Manage go module dependencies (#148)

dependabot[bot] committed:
Bump actions/cache from 2 to 3 (#149)

Bumps [actions/cache](https://github.com/actions/cache) from 2 to 3.
- [Release notes](https://github.com/actions/cache/releases)
- [Commits](https://github.com/actions/cache/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/cache
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

svc-excavator-bot committed:
Excavator: Update godel to latest version (#150)

asvoboda committed:
Remove dependency on go-baseapp (#152)

dependabot[bot] committed:
Bump actions/setup-go from 2 to 3 (#153)

Bumps [actions/setup-go](https://github.com/actions/setup-go) from 2 to 3.
- [Release notes](https://github.com/actions/setup-go/releases)
- [Commits](https://github.com/actions/setup-go/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/setup-go
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

svc-excavator-bot committed:
Excavator: Manage go version (#151)

dependabot[bot] committed:
Bump github/codeql-action from 1 to 2 (#155)

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 1 to 2.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/v1...v2)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

svc-excavator-bot committed:
Excavator: Manage go version (#156)

svc-excavator-bot committed:
Excavator: Updates godel plugins to latest versions (#154)

svc-excavator-bot committed:
Excavator: Update godel to latest version (#157)

svc-excavator-bot committed:
Excavator: Manage go module dependencies (#158)

svc-excavator-bot committed:
Excavator: Manage go version (#160)

svc-excavator-bot committed:
Excavator: Update godel to latest version (#161)

svc-excavator-bot committed:
Excavator: Updates godel plugins to latest versions (#162)

svc-excavator-bot committed:
Excavator: Manage go module dependencies (#164)

svc-excavator-bot committed:
Excavator: Update godel to latest version (#163)

svc-excavator-bot committed:
Excavator: Updates godel plugins to latest versions (#165)

asvoboda committed:
Fix permissions for codeql (#166)

bluekeyes committed:
Update go-github to v45.2.0 (#167)

asvoboda committed:
Update ghinstallation to 2.1.0 (#168)

svc-excavator-bot committed:
Excavator: Manage go version (#171)

svc-excavator-bot committed:
Excavator:  Update godel to latest version (#173)

svc-excavator-bot committed:
Excavator:  Manage go version (#175)

svc-excavator-bot committed:
Excavator:  Updates godel plugins to latest versions (#174)

dependabot[bot] committed:
Bump github.com/rs/zerolog from 1.27.0 to 1.28.0 (#178)

Bumps [github.com/rs/zerolog](https://github.com/rs/zerolog) from 1.27.0 to 1.28.0.
- [Release notes](https://github.com/rs/zerolog/releases)
- [Commits](https://github.com/rs/zerolog/compare/v1.27.0...v1.28.0)

---
updated-dependencies:
- dependency-name: github.com/rs/zerolog
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

steebchen committed:
Add WithRedirectURL option  (#177)

svc-excavator-bot committed:
Excavator:  Updates godel plugins to latest versions (#179)

svc-excavator-bot committed:
Excavator:  Update godel to latest version (#180)

svc-excavator-bot committed:
Excavator:  Updates godel plugins to latest versions (#181)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#182)

bluekeyes committed:
Update go-github to v47.0.0 (#183)

svc-excavator-bot committed:
Excavator:  Update godel to latest version (#184)

svc-excavator-bot committed:
Excavator:  Manage go version (#185)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#186)

dependabot[bot] committed:
Bump github.com/google/go-github/v47 from 47.0.0 to 47.1.0 (#187)

Bumps [github.com/google/go-github/v47](https://github.com/google/go-github) from 47.0.0 to 47.1.0.
- [Release notes](https://github.com/google/go-github/releases)
- [Commits](https://github.com/google/go-github/compare/v47.0.0...v47.1.0)

---
updated-dependencies:
- dependency-name: github.com/google/go-github/v47
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

svc-excavator-bot committed:
Excavator:  Manage go version (#188)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#189)

svc-excavator-bot committed:
Excavator:  Manage go version (#190)

svc-excavator-bot committed:
Excavator:  Updates godel plugins to latest versions (#191)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#192)

svc-excavator-bot committed:
Excavator:  Update godel to latest version (#193)

svc-excavator-bot committed:
Excavator:  Updates godel plugins to latest versions (#196)

svc-excavator-bot committed:
Excavator:  Manage go version (#195)

svc-excavator-bot committed:
Excavator:  Update godel to latest version (#194)

svc-excavator-bot committed:
Excavator:  Update policy-bot config (#198)

svc-excavator-bot committed:
Excavator:  Render Bulldozer Bot config file (#197)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#199)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#200)

svc-excavator-bot committed:
Excavator:  Update godel to latest version (#201)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#202)

asvoboda committed:
Create excavator config file (#204)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#205)

dependabot[bot] committed:
Bump github.com/hashicorp/golang-lru from 0.5.4 to 0.6.0 (#206)

Bumps [github.com/hashicorp/golang-lru](https://github.com/hashicorp/golang-lru) from 0.5.4 to 0.6.0.
- [Release notes](https://github.com/hashicorp/golang-lru/releases)
- [Commits](https://github.com/hashicorp/golang-lru/compare/v0.5.4...v0.6.0)

---
updated-dependencies:
- dependency-name: github.com/hashicorp/golang-lru
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#207)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#208)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#209)

svc-excavator-bot committed:
Excavator:  Manage go version (#210)

svc-excavator-bot committed:
Excavator:  Updates godel plugins to latest versions (#211)

asvoboda committed:
Update go-github to v48 (#212)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#215)

svc-excavator-bot committed:
Excavator:  Update godel to latest version (#217)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#218)

asvoboda committed:
Update go-github to v50 (#219)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#220)

svc-excavator-bot committed:
Excavator:  Update godel to latest version (#221)

svc-excavator-bot committed:
Excavator:  Updates godel plugins to latest versions (#222)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#223)

svc-excavator-bot committed:
Excavator:  Update godel to latest version (#224)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#225)

svc-excavator-bot committed:
Excavator:  Manage go version (#226)

svc-excavator-bot committed:
Excavator:  Manage go version (#228)

svc-excavator-bot committed:
Excavator:  Updates godel plugins to latest versions (#229)

svc-excavator-bot committed:
Excavator:  Manage go version (#232)

dependabot[bot] committed:
Bump github.com/google/go-github/v50 from 50.0.0 to 50.1.0 (#230)

Bumps [github.com/google/go-github/v50](https://github.com/google/go-github) from 50.0.0 to 50.1.0.
- [Release notes](https://github.com/google/go-github/releases)
- [Commits](https://github.com/google/go-github/compare/v50.0.0...v50.1.0)

---
updated-dependencies:
- dependency-name: github.com/google/go-github/v50
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump golang.org/x/oauth2 from 0.0.0-20210113205817-d3ed898aa8a3 to 0.5.0 (#231)

Bumps [golang.org/x/oauth2](https://github.com/golang/oauth2) from 0.0.0-20210113205817-d3ed898aa8a3 to 0.5.0.
- [Release notes](https://github.com/golang/oauth2/releases)
- [Commits](https://github.com/golang/oauth2/commits/v0.5.0)

---
updated-dependencies:
- dependency-name: golang.org/x/oauth2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

svc-excavator-bot committed:
Excavator:  Update godel to latest version (#233)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#234)

svc-excavator-bot committed:
Excavator:  Update godel to latest version (#235)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#236)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#237)

dependabot[bot] committed:
Bump golang.org/x/oauth2 from 0.5.0 to 0.6.0 (#238)

svc-excavator-bot committed:
Excavator:  Update godel to latest version (#239)

svc-excavator-bot committed:
Excavator:  Update policy-bot config (#240)

dependabot[bot] committed:
Bump github.com/google/go-github/v50 from 50.1.0 to 50.2.0 (#242)

Bumps [github.com/google/go-github/v50](https://github.com/google/go-github) from 50.1.0 to 50.2.0.
- [Release notes](https://github.com/google/go-github/releases)
- [Commits](https://github.com/google/go-github/compare/v50.1.0...v50.2.0)

---
updated-dependencies:
- dependency-name: github.com/google/go-github/v50
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump actions/setup-go from 3 to 4 (#241)

Bumps [actions/setup-go](https://github.com/actions/setup-go) from 3 to 4.
- [Release notes](https://github.com/actions/setup-go/releases)
- [Commits](https://github.com/actions/setup-go/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/setup-go
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

svc-excavator-bot committed:
Excavator:  Manage go version (#243)

svc-excavator-bot committed:
Excavator:  Updates godel plugins to latest versions (#244)

svc-excavator-bot committed:
Excavator:  Update godel to latest version (#245)

svc-excavator-bot committed:
Excavator:  Manage go version (#246)

svc-excavator-bot committed:
Excavator:  Updates godel plugins to latest versions (#247)

dependabot[bot] committed:
Bump github.com/bradleyfalzon/ghinstallation/v2 from 2.2.0 to 2.3.0 (#250)

Bumps [github.com/bradleyfalzon/ghinstallation/v2](https://github.com/bradleyfalzon/ghinstallation) from 2.2.0 to 2.3.0.
- [Release notes](https://github.com/bradleyfalzon/ghinstallation/releases)
- [Commits](https://github.com/bradleyfalzon/ghinstallation/compare/v2.2.0...v2.3.0)

---
updated-dependencies:
- dependency-name: github.com/bradleyfalzon/ghinstallation/v2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump golang.org/x/oauth2 from 0.6.0 to 0.7.0 (#249)

Bumps [golang.org/x/oauth2](https://github.com/golang/oauth2) from 0.6.0 to 0.7.0.
- [Release notes](https://github.com/golang/oauth2/releases)
- [Commits](https://github.com/golang/oauth2/compare/v0.6.0...v0.7.0)

---
updated-dependencies:
- dependency-name: golang.org/x/oauth2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#251)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#252)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#254)

asvoboda committed:
Update go-github to v52 (#253)

svc-excavator-bot committed:
Excavator:  Update policy-bot config (#255)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#256)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#257)

dependabot[bot] committed:
Bump github.com/cloudflare/circl from 1.1.0 to 1.3.3 (#258)

Bumps [github.com/cloudflare/circl](https://github.com/cloudflare/circl) from 1.1.0 to 1.3.3.
- [Release notes](https://github.com/cloudflare/circl/releases)
- [Commits](https://github.com/cloudflare/circl/compare/v1.1.0...v1.3.3)

---
updated-dependencies:
- dependency-name: github.com/cloudflare/circl
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

svc-excavator-bot committed:
Excavator:  Update godel to latest version (#259)

svc-excavator-bot committed:
Excavator:  Manage go version (#260)

svc-excavator-bot committed:
Excavator:  Update godel to latest version (#261)

svc-excavator-bot committed:
Excavator:  Updates godel plugins to latest versions (#262)

asvoboda committed:
Update go-github to v53 (#263)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#264)

dependabot[bot] committed:
Bump github.com/google/go-github/v53 from 53.0.0 to 53.1.0 (#265)

svc-excavator-bot committed:
Excavator:  Manage go version (#266)

dependabot[bot] committed:
Bump github.com/google/go-github/v53 from 53.1.0 to 53.2.0 (#267)

Bumps [github.com/google/go-github/v53](https://github.com/google/go-github) from 53.1.0 to 53.2.0.
- [Release notes](https://github.com/google/go-github/releases)
- [Commits](https://github.com/google/go-github/compare/v53.1.0...v53.2.0)

---
updated-dependencies:
- dependency-name: github.com/google/go-github/v53
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

nvelat committed:
add a caching implementation for InstallationsService (#268)

* add caching installation service

* add license header

* typo in sprintf

* switch to time based cache

svc-excavator-bot committed:
Excavator:  Updates godel plugins to latest versions (#269)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#270)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#271)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#272)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#274)

svc-excavator-bot committed:
Excavator:  Manage go version (#275)

svc-excavator-bot committed:
Excavator:  Updates godel plugins to latest versions (#276)

svc-excavator-bot committed:
Excavator:  Update policy-bot config (#277)

svc-excavator-bot committed:
Excavator:  Update godel to latest version (#273)

svc-excavator-bot committed:
Excavator:  Update godel to latest version (#278)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#279)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#280)

svc-excavator-bot committed:
Excavator:  Manage go version (#281)

svc-excavator-bot committed:
Excavator:  Updates godel plugins to latest versions (#282)

svc-excavator-bot committed:
Excavator:  Update godel to latest version (#283)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#284)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#285)

svc-excavator-bot committed:
Excavator:  Update godel to latest version (#286)

svc-excavator-bot committed:
Excavator:  Manage go version (#287)

asvoboda committed:
Update go-github to v55 (#288)

svc-excavator-bot committed:
Excavator:  Updates godel plugins to latest versions (#289)

dependabot[bot] committed:
Bump actions/checkout from 3 to 4 (#290)

Bumps [actions/checkout](https://github.com/actions/checkout) from 3 to 4.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#291)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#292)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#293)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#294)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#295)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#296)

mortenmj committed:
Update google/go-github to v56 (#297)

svc-excavator-bot committed:
Excavator:  Manage go version (#298)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#299)

svc-excavator-bot committed:
Excavator:  Updates godel plugins to latest versions (#300)

svc-excavator-bot committed:
Excavator:  Update godel to latest version (#301)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#302)

svc-excavator-bot committed:
Excavator:  Manage go version (#303)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#304)

svc-excavator-bot committed:
Excavator:  Updates godel plugins to latest versions (#305)

svc-excavator-bot committed:
Excavator:  Manage go module dependencies (#306)

dheerajodha committed:
Ensure that the app updates the comment on PR by a target user

dheerajodha commented on example/issue_comment.go:
```diff
@@ -66,22 +82,98 @@ func (h *PRCommentHandler) Handle(ctx context.Context, eventType, deliveryID str
 	return nil
 }
+
+func getFailedTestCases(scanner *prow.ArtifactScanner, logger zerolog.Logger) []string {
```
One missing thing is to cover the case when the given prow URL fails during the infra-setup.

dheerajodha replied:
Addressed with https://github.com/dheerajodha/go-githubapp/pull/1/commits/7e350f841f026d0c0b649ddc5affaed874b9e922

psturc commented on example/issue_comment.go:
```diff
@@ -66,22 +82,98 @@ func (h *PRCommentHandler) Handle(ctx context.Context, eventType, deliveryID str
+	failedTestCasesNames := []string{}
+
+	overallJUnitSuites := &reporters.JUnitTestSuites{}
+	openshiftCiJunit := reporters.JUnitTestSuite{Name: "openshift-ci job", Properties: reporters.JUnitProperties{Properties: []reporters.JUnitProperty{}}}
```
"openshift-ci job" - I think it would be good if this is a constant

@dheerajodha would you be ok with reflecting it in [related qe-tools PR](https://github.com/redhat-appstudio/qe-tools/pull/42/files#diff-e776dc4867d14d4e0aba7961297933a914522b6dda35231f9a4d528320807580L64) please?

dheerajodha replied:
I've removed this LOC because it didn't make much sense to have it in the first place.
But added "openshift-ci job" as a constant. Thanks!

psturc commented on example/issue_comment.go:
```diff
@@ -17,13 +17,29 @@ package main
+
+const (
+	targetAuthor  = "dheerajodha"
+	buildLogFilename = "build-log.txt"
+	finishedFilename = "finished.json"
```
are these constants used anywhere?

dheerajodha replied:
nope, removed them.

psturc commented on example/issue_comment.go:
```diff
@@ -17,13 +17,29 @@ package main
+	finishedFilename = "finished.json"
+	junitFilename = `/(j?unit|e2e).*\.xml`
+
+	gcsBrowserURLPrefix = "https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/"
```
this one seems unused as well

dheerajodha replied:
removed as well.

psturc commented on example/issue_comment.go:
```diff
@@ -17,13 +17,29 @@ package main
+)
+
+const (
+	targetAuthor  = "dheerajodha"
```
I guess this will change after we move the repo to redhat-appstudio org

dheerajodha replied:
Yes, planning to do this as soon I get the confirmation that the rest of the code is ready to get merged.

psturc reviewed (commented):
just couple of comments, I think this is good for a start 👍

dheerajodha committed:
feat(KONFLUX-782): also process the prow jobs with setup failures

This commit ensures that the GH App also processes the prow jobs
which fails on setting up the infrastructure. We surface this failure
back to the PR to let the users know that the latest job failed before
it could even execute the e2e-tests.

Signed-off-by: Dheeraj<djodha@redhat.com>

dheerajodha committed:
feat(KONFLUX-782): some code refactoring and minor changes

* Introduce a struct FailedTestCasesReport, to easily diffrentiate the job
  failures during cluster bootstrapping phase, and E2E tests execution phase.
* Make the code more modular by moving some LOCs into separate functions.
* Better logging to make debugging easier.

Signed-off-by: Dheeraj<djodha@redhat.com>

dheerajodha committed:
feat(KONFLUX-782): deal with prow jobs that fail while cluster creation

* handle jobs that failed during cluster creation and display the info
  back to the user.
* add a new Failure/Error message for each failed test case displayed to the
  PR's comment.

Singed-off-by: Dheeraj<djodha@redhat.com>

dheerajodha commented on example/issue_comment.go:
```diff
@@ -42,46 +65,187 @@ func (h *PRCommentHandler) Handle(ctx context.Context, eventType, deliveryID str
+func setHeaderString(logger zerolog.Logger, overallJUnitSuites *reporters.JUnitTestSuites) *FailedTestCasesReport {
+	failedTCReport := FailedTestCasesReport{}
+
+	if len(overallJUnitSuites.TestSuites) == 0 {
+		logger.Debug().Msg("The given Prow job failed while creating the cluster")
+		failedTCReport.headerString = ":rotating_light: **Error occurred while creating the cluster, please check the Prow's build logs.**\n"
+	} else if len(overallJUnitSuites.TestSuites) == 1 && overallJUnitSuites.TestSuites[0].Name == openshiftCITestSuiteName {
+		logger.Debug().Msg("The given Prow job failed during bootstrapping the cluster")
+		failedTCReport.hasBootstrapFailure = true
+		failedTCReport.headerString = ":rotating_light: **Error occurred during the cluster's Bootstrapping phase, list of failed Spec(s)**: \n"
+	} else {
+		logger.Debug().Msg("The given Prow job failed while running the E2E tests")
+		failedTCReport.headerString = ":rotating_light: **Error occurred while running the E2E tests, list of failed Spec(s)**: \n"
+	}
```
Here's how the Prow job for these 3 `if`, `else if`, and `else` would be processed by the GitHub and displayed to the users:
* [Example](https://github.com/dheerajodha/sample-github-app/pull/5#issuecomment-1898171606), when the Prow job failed during cluster creation.
* [Example](https://github.com/dheerajodha/sample-github-app/pull/5#issuecomment-1898175959),  when the Prow job failed while bootstrapping (or installing Konflux on) the cluster -- lmk if I should change the phrasing.
* When the Prow job failed while running the E2E tests:
** [where the E2E test](https://github.com/dheerajodha/sample-github-app/pull/5#issuecomment-1898180007) **failed**
** [where the E2E test](https://github.com/dheerajodha/sample-github-app/pull/5#issuecomment-1898177916) **panicked**

Dannyb48 replied:
@psturc @dheerajodha For the example when the prow job failed while boostrapping the cluster, what are your thoughts on looking at the system-error element of that test case and showing like the last 12-15 lines? Rather than looking at the failure-message element? I didn't really find that message of `redhat-appstudio-e2e has failed` helpful for that use case.

psturc replied:
yep, I think that is a good idea

Dannyb48 replied:
Are we going to include the changes I mentioned here as part of this PR or no?

dheerajodha replied:
I'll include them as part of this PR, thanks for the suggestion!

dheerajodha replied:
Addressed with [this commit](https://github.com/dheerajodha/go-githubapp/pull/1/commits/e9250f3633acda69976b66d49f5df6c81d929a16).

Here's how the updated comment now looks like: https://github.com/dheerajodha/sample-github-app/pull/5#issuecomment-1906841527

psturc reviewed (commented):
lgtm, I think it's good to merge after [merging required changes in qe-tools](https://github.com/redhat-appstudio/qe-tools/pull/42)

dheerajodha committed:
feat(KONFLUX-782): in case of bootstrap failure, display last few lines

* With this change, if a prow job fails during bootstrapping Konflux to cluser
  then the GitHub app display last few lines of the error log, to the comment.
* This will provide more info to the users about what went wrong.

Signed-off-by: Dheeraj<djodha@redhat.com>

dheerajodha committed:
feat(KONFLUX-782): change the target author to OpenShift-CI bot

Signed-off-by: Dheeraj<djodha@redhat.com>

dheerajodha commented:
Closed this PR since these changes found a new home here: https://github.com/redhat-appstudio-qe/ci-helper-app/pull/1
